inc/Tecs_lock.hh

```diff
@@ -52,7 +52,8 @@ namespace Tecs {
 
     public:
         // Start a new transaction
-        inline Lock(ECS &instance) : instance(instance), base(new Transaction<ECS, Permissions...>(instance)) {
+        inline Lock(ECS &instance) : instance(instance) {
+            base = std::make_shared<Transaction<ECS, Permissions...>>(instance);
```

Comment:
This change just puts the shared_ptr state and the Transaction in the same allocation.